source/crud/bulk-write.md

```diff
@@ -0,0 +1,778 @@
+    "multi": Optional<Boolean>,
+    "upsert": Optional<Boolean>,
+    "arrayFilters": Optional<Array>,
+    "hint": Optional<Document | String>
+}
+```
+
+#### Delete
+
+```json
+{
+    "delete": <Int32>,
+    "filter": <Document>,
+    "multi": Optional<Boolean>,
+    "hint": Optional<Document | String>,
+    "collation": Optional<Document>
+}
+```
+
+### Namespace Information
+
+The `nsInfo` field is an array containing the namespaces on which the write operations should be
+performed. Drivers MUST NOT include duplicate namespaces in this list. The documents in the
+`nsInfo` array have the following format:
+
+```json
+{
+    "ns": <String>
+}
+```
+
+### `errorsOnly` and `verboseResults`
+
+The `errorsOnly` field indicates whether the results cursor returned in the `bulkWrite` response
+should contain only errors and omit individual results. If false, both individual results for
+successful operations and errors will be returned. This field is optional and defaults to false on
+the server.
+
+`errorsOnly` corresponds to the `verboseResults` option defined on `BulkWriteOptions`. If the user
+specified a value for `verboseResults`, drivers MUST define `errorsOnly` as the opposite of
+`verboseResults`. If the user did not specify a value for `verboseResults`, drivers MUST define
+`errorsOnly` as `true`.
+
+### `ordered`
+
+The `ordered` field defines whether writes should be executed in the order in which they were
+specified, and, if an error occurs, whether the server should halt execution of further writes. It
+is optional and defaults to true on the server. Drivers MUST explicitly define `ordered` as `true`
+in the `bulkWrite` command if a value is not specified in `BulkWriteOptions`. This is required to
+avoid inconsistencies between server and driver behavior if the server default changes in the
+future.
+
+### Size Limits
+
+The server reports a `maxBsonObjectSize` in its `hello` response. This value defines the maximum
+size for documents that are inserted into the database. Documents that are sent to the server but
+are not intended to be inserted into the database (e.g. command documents) have a size limit of
+`maxBsonObjectSize + 16KiB`. When an acknowledged write concern is used, drivers MUST NOT perform
+any checks related to these size limits and MUST rely on the server to raise an error if a limit is
+exceeded. However, when an unacknowledged write concern is used, drivers MUST raise an error if one
+of the following limits is exceeded:
+
+- The size of a document to be inserted MUST NOT exceed `maxBsonObjectSize`. This applies to the
+  `document` field of an `InsertOneModel` and the `replacement` field of a `ReplaceOneModel`.
+- The size of an entry in the `ops` array MUST NOT exceed `maxBsonObjectSize + 16KiB`.
+- The size of the `bulkWrite` command document MUST NOT exceed `maxBsonObjectSize`. The overhead
+  bytes are omitted here intentionally to accommodate additional fields that mongos appends to the
+  command document.
```

Comment:
```suggestion
- The size of the `bulkWrite` command document MUST NOT exceed `maxBsonObjectSize + 16KiB`.
```

I prefer the driver only error when the the operation is certain not to succeed. Otherwise rely on the server to error.

When using unacknowledged writes, I expect there will be many cases where a write is likely to error, but a driver cannot reasonably check (e.g. server document validation). When in doubt, I rather err towards having the driver send what the user sets.

Reply:
that's reasonable, updated